snap/info.go

```diff
@@ -270,17 +267,12 @@ func (s *Info) DataDir() string {
 
 // UserDataDir returns the user-specific data directory of the snap.
 func (s *Info) UserDataDir(home string) string {
-	return filepath.Join(home, "snap", s.Name(), s.Revision.String())
-}
-
-// HomeDirBase returns the user-specific home directory base of the snap.
-func (s *Info) HomeDirBase(home string) string {
-	return filepath.Join(home, "snap", s.Name())
+	return filepath.Join(dirs.GlobalRootDir, home, "snap", s.Name(), s.Revision.String())
```

Comment:
Adding the rootdir prefix here feels a bit strange since we're getting a path as the function input. People calling this won't expect the path to be relative I believe, which means we might see bugs happening due to doubled paths when the root dir is not /. Imagine someone iterating over /home, for example.. people would most likely not strip half the path out.